tests/components/transmission/test_init.py

```diff
@@ -73,3 +77,66 @@ async def test_unload_entry(hass: HomeAssistant) -> None:
 
     assert entry.state is ConfigEntryState.NOT_LOADED
     assert not hass.data[DOMAIN]
+
+
+@pytest.mark.parametrize(
+    ("domain", "old_unique_id", "new_unique_id"),
+    [
+        (SENSOR_DOMAIN, "0.0.0.0-Transmission Down Speed", "1234-download"),
+        (SENSOR_DOMAIN, "0.0.0.0-Transmission Up Speed", "1234-upload"),
+        (SENSOR_DOMAIN, "0.0.0.0-Transmission Status", "1234-status"),
+        (
+            SENSOR_DOMAIN,
+            "0.0.0.0-Transmission Active Torrents",
+            "1234-active_torrents",
+        ),
+        (
+            SENSOR_DOMAIN,
+            "0.0.0.0-Transmission Paused Torrents",
+            "1234-paused_torrents",
+        ),
+        (SENSOR_DOMAIN, "0.0.0.0-Transmission Total Torrents", "1234-total_torrents"),
+        (
+            SENSOR_DOMAIN,
+            "0.0.0.0-Transmission Completed Torrents",
+            "1234-completed_torrents",
+        ),
+        (
+            SENSOR_DOMAIN,
+            "0.0.0.0-Transmission Started Torrents",
+            "1234-started_torrents",
+        ),
+        (SENSOR_DOMAIN, "0.0.0.0-download", "0.0.0.0-download"),
+        (SENSOR_DOMAIN, "abcde", "abcde"),
+        (SWITCH_DOMAIN, "0.0.0.0-Transmission Switch", "1234-on_off"),
+        (SWITCH_DOMAIN, "0.0.0.0-Transmission Turtle Mode", "1234-turtle_mode"),
+        (SWITCH_DOMAIN, "0.0.0.0-on_off", "0.0.0.0-on_off"),
+        (SWITCH_DOMAIN, "abcde", "abcde"),
+    ],
+)
+async def test_migrate_unique_id(
+    hass: HomeAssistant, domain: str, old_unique_id: str, new_unique_id: str
```

Comment:
You can use the fixture here:
```suggestion
    hass: HomeAssistant,
    entity_registry: er.EntityRegistry,
    domain: str,
    old_unique_id: str,
    new_unique_id: str,
```